File: daily/1448-Count-good-nodes-in-binary-tree.go
package daily

var res int = 0

func goodNodes(root *TreeNode) int {
	res = 0
	dfsGoodNodes(root, root.Val)
	return res
}

func dfsGoodNodes(node *TreeNode, max_val int) {
	if node == nil {
		return
	}
	if node.Val >= max_val {
		res += 1
		max_val = node.Val
	}
	dfsGoodNodes(node.Left, max_val)
	dfsGoodNodes(node.Right, max_val)
}

File: daily/Binary Tree/1448-Count-good-nodes-in-binary-tree.py
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.res = 0
        def dfs(node, maxValue):
            if not node:
                return
            if node.val >= maxValue:
                self.res += 1
                maxValue = node.val
            dfs(node.left, maxValue)
            dfs(node.right, maxValue)
        dfs(root, root.val)
        return self.res